lib/Telemetry/src/DispatcherFactory.h

```diff
@@ -0,0 +1,114 @@
+#include "DataQueue.h"
+#include "LogCommand.h"
+#include "Command.h"
+#include "Dispatcher.h"
+#include "Particle.h"
+#include "IntervalLogger.h"
+
+#define MAX_NUM_INTERVALS 5
+
+#ifndef _DISPTACHER_FACTORY_H_
+#define _DISPATCHER_FACTORY_H_
+
+// This isn't really a factory but it has a similar purpose:
+// Add all the commands you want to this and call build().  It
+// produces a dispatcher with all logging functionality already set up
+// You can delete this safely after build() has been called
+class DispatcherFactory {
+    public:
+        /**
+         * Constructs new DispatcherFactory with number of commands and data queue
+         * 
+         * @param numCommands The number of sensor functions you will add to the factory
+         * Note: this must be accurate and has potential to cause a segfault if the value is too low
+         * @param dataQ the Data Queue from main
+         * */
+        DispatcherFactory(uint16_t numCommands, DataQueue *dataQ) {
+            _dataQ = dataQ;
+            _commands = new Command*[numCommands];
+            _intervals = new uint16_t[MAX_NUM_INTERVALS] ();
+            _numCommandsOnIntervals = new uint16_t[MAX_NUM_INTERVALS] ();
+            _numCommands = 0;
+            _numIntervals = 0;
+        }
+        
+        /**
+         * Destructor which frees unused heap data (can safely be called after build)
+        **/
+        ~DispatcherFactory() {
+            delete _intervals;
+            delete _numCommandsOnIntervals;
+        }
+
+        /**
+         * Adds a Sensor-property getter function, the property name, and interval at which to call it
+         * 
+         * @param S the address of the Sensor whose getter you want to be called
```

Comment:
I think this first param should be "sensor". Can you also add comments about what "S" and "R" should be in this context?

Reply:
yes, I noticed this mistake after I committed.  It has now been changed